src/collections/_documentation/performance/distributed-tracing.md

```diff
@@ -12,105 +12,325 @@ Sentry's Performance features are currently in beta. For more details about acce
 
-If you enable Performance collection for a large portion of your traffic, you may exceed your organization's [Quotas and Rate Limits]({%- link _documentation/accounts/quotas/index.md -%}).
+**Orphan Transactions**
 
-## Viewing Transactions
+In theory, in a fully instrumented system, each trace should contain only one transaction and one span (the transaction's root) without a parent, namely the transaction in the originating service. However, in practice, you may not have tracing enabled in every one of your services, or an instrumented service may fail to report a transaction due to network disruption or other unforeseen circumstances. When this happens, you may see gaps in your trace hierarchy. Specifically, you may see transactions partway through the trace whose parent spans haven't been recorded as part of any known transactions. Such non-originating, parentless transactions are called **orphan transactions**.
 
-View transaction events by clicking on the "Transactions" pre-built query in [Discover]({%- link _documentation/performance/discover/index.md -%}), or by using a search condition `event.type:transaction` in a [Discover Query Builder]({%- link _documentation/performance/discover/query-builder.md -%}) view.
+**Nested Spans**
 
-When you open a transaction event in Discover, you'll see the **span view** at the top of the page. Other information the SDK captured as part of the Transaction event, such as breadcrumbs, will also be displayed.
+Though our examples above had four levels in their hierarchy (trace, transaction, span, child span) there's no set limit to how deep the nesting of spans can go. There are, however, practical limits: transaction payloads sent to Sentry have a maximum allowed size, and as with any kind of logging, there's a balance to be struck between your data's granularity and its usability. 
 
-[{% asset performance/discover-span.png alt="Discover span showing the map of the transactions (aka minimap) and the black dashed handlebars for adjusting the window selection." %}]({% asset performance/discover-span.png @path %})
+**Zero-duration Spans**
+
+It's possible for a span to have equal start and end times, and therefore be recorded as taking no time. This can occur either because the span is being used as a marker (such as is done in [the browser's Performance API](https://developer.mozilla.org/en-US/docs/Web/API/Performance/mark)) or because the amount of time the operation took is less than the measurement resolution (which will vary by service).
+
+**Clock Skew**
```

Comment:
Thanks! 😊 

As for the timestamps:

Even in the most manual cases of users collecting their own spans, my understanding is that the start and end timestamps are still "private" fields. In other words, I would think that the only ways for this to happen are for the computer's clock to be running backwards (unlikely, but you never know...), or for users to muck with the data in `beforeSend` (at which point it's on them, IMHO).

Are there languages where you can anticipate that not being the case, though? Where we'll be forced to have users set the timestamps themselves, rather than relying on whatever that language's version of `DateTime.now()` is? If so... then I'm still not sure it's not too much in the weeds, but I'd be willing to be convinced. 😛

Reply:
> my understanding is that the start and end timestamps are still "private" fields

Hmm, neither Python nor JS are good examples because nothing is really private in those languages. And that's where we have transactions so far.

Sometimes it is interesting that those fields are public and settable -- consider when the measurement was done by some external process and you want to add a Span to the transaction. That's exactly what Abhijeet is working on now, a OpenTelemetry exporter. It will construct Sentry spans setting start and end times coming from another OTel-compatible system.

> I would think that the only ways for this to happen are for the computer's clock to be running backwards (unlikely, but you never know...)

That's the very reason I went on to retrofit using monotonic clocks in both the Python and JS SDKs... Computer clocks are not a perfect mathematical model.

`datetime.now()` and `new Date()` are NOT monotonic, they can and do change, including moving backwards and forwards by arbitrary amounts.

Try changing your computer clock in the middle of a span by going back to 2019. And before we say "users will not do that", the OS itself can change the clock in response to NTP, bad hardware clock, etc. VMs are also known to have clock skew problems (and they are heavily used in "the cloud", where everyone's code runs).

> Are there languages where you can anticipate that not being the case, though? 

Not all languages have a monotonic clock available.
Even our current Python implementation doesn't use a monotonic clock today on Python 2.x. So it could happen "by accident".


I agree with you that SDKs should handle this as much as they can and perhaps this doesn't need to be documented, or at least not now. I saw the other topics, thought about  this timestamp precondition and commented here.

Could also end up in the more technical SDK development docs, if not there already.

Reply:
Man, TIL. Time really _can_ flow backwards!

And yeah, it's true that neither Python nor JS have truly private fields, i.e., we can't enforce users not setting the data if they're determined to do so. I tend to have less sympathy for people who actively work against the given API, but you raise valid points about the possibility of the timestamps being wrong even if the SDK sets them.

Nothing is ever simple, is it? Someday I will learn...

In any case, given all of the above, I think you're right, we should document it, but I do think the developer docs are probably(?) a better place. @HazAT, what do you think?

Reply:
Talked to @HazAT offline - we will mention the "you gotta begin before you end" bit in the dev docs.